### PR TITLE
Example of an igrins2 observation simulator

### PR DESCRIPTION
**Pull request by cquiroz**
This is an initial sample of simulating an IGRINS2 observation.
It emulates sending the observation events as the instrument and also start/end events as the seqexec
It is limited as of now not adding the full set of seqexec keywords nor status item keywords.

However paired with GDS, it emulates enough functionality to properly test the observation execution chain

### CODE REVIEW

File: src/examples/Makefile [hunk 1]
```diff
@@ -10,7 +10,7 @@ LIBS= -lgiapi-glue-cc -lactivemq-cpp -llog4cxx -lapr-1 -lcurlpp -lcurl
 
 LD_LIBRARY_PATH := ../../:$(LOG4CXX_LIB):$(CPPUNIT_LIB):$(ACTIVEMQ_LIB):$(APR_LIB):$(CURLPP_LIB)
 
-all: seq-command tcscontext pcsupdater obsevent epicssubscription epicsget logging fileevents status services apply-command any-command observe-simulator 
+all: seq-command tcscontext pcsupdater obsevent epicssubscription epicsget logging fileevents status services apply-command any-command observe-simulator
```
Comment (framos-dev) [inline src/examples/Makefile]: Shouldn't igrins2obssimulator be added so that the binary is created?

File: src/examples/Makefile [hunk 2]
```diff
@@ -78,9 +81,9 @@ epicssubscription: epicssubscription.o EpicsHandlerDemo.o
 %: %.o
 	$(CXX) $(LIB_DIRS) $(LIBS) -o $@  $<
 
-run: 
+run:
```
Comment (framos-dev) [inline src/examples/Makefile]: Do we really want to add this target? if we want to add it we should also put it in the "all" or we would add it when we want to see the LD_LIBRARY_PATH? It looks like more testing.

**Comment by framos-gemini** [comment]
thank you @cquiroz  for making this PR with the example of src/examples/igrins2obssimulator.cpp and marking the first example after the new GDS implementation.

Taking advantage of the above comment, the examples from before this change are still valid (gpiobssimulator.cpp, gpisimulator.cpp, and obshighcount.cpp)?

**Comment by cquiroz** [comment]
Good question, theyy are valid for gmp R14 as used by ghost
But maybe we should delete them

**Comment by framos-gemini** [comment]
it's perfect for me, thanks @cquiroz

**Review by framos-gemini — APPROVED** [review]

**Review by framos-gemini — APPROVED** [review]

**Review by framos-gemini — APPROVED** [review]

File: src/examples/Makefile (hunk 1 above)
Comment (cquiroz) [reply src/examples/Makefile]: Added

File: src/examples/Makefile (hunk 2 above)
Comment (cquiroz) [reply src/examples/Makefile]: Let's remove it

**Pull request merged**